CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+
+        public IJokeService JokeService { get; }
+        public IOutputService OutputService { get; }
+
+        public Jester(IOutputService? outputService, IJokeService? jokeService)
+        {
+            JokeService = jokeService ?? throw new ArgumentNullException(nameof(jokeService));
+            OutputService = outputService ?? throw new ArgumentNullException(nameof(outputService));
+        }

```

Comment:
```suggestion
        public Jester(IOutputService outputService, IJokeService jokeService)
        {
            JokeService = jokeService ?? throw new ArgumentNullException(nameof(jokeService));
            OutputService = outputService ?? throw new ArgumentNullException(nameof(outputService));
        }
```

Null checks are a valid use-case even on non-nullable members.
https://stevetalkscode.co.uk/null-injection
We shouldn't trick users into passing in nullable services when doing so will throw an exception immediately.